bin/orphaned-pr-environments

```
@@ -17,7 +19,7 @@ echo "::endgroup::"
 echo "::group::List PRs with PR environments"
 echo terraform -chdir="infra/${app_name}/service" workspace list
 workspaces="$(terraform -chdir="infra/${app_name}/service" workspace list)"
-pr_nums="$(echo "${workspaces}" | grep -o 'p-[0-9]\+' | sed 's/p-//')"
+pr_nums="$(echo "${workspaces}" | { grep -o 'p-[0-9]\+' || test $? = 1; } | sed 's/p-//')"
```

can we add a comment like

```
# Search for workspaces that start with "p-" and extract the PR numbers
# The "test $? = 1" is used to avoid errors if there are no PR environments
```

also, can you help me understand the advantage of `test $? = 1` over `|| true`? The latter seems simpler to grok.

---

`|| true` ignores all errors, which while unlikely seems worse than just ignoring the specific expected case. grep will exit with `2` if something else is wrong.

---

ah ok that makes sense, thanks